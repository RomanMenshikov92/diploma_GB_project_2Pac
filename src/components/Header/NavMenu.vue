<template lang="pug">
nav.header__nav
  ul.list-reset.header__list
    li(
      v-for="(item, index) in menu"
      :key="index"
      class="header__item"
      :class="{'header__item--active': isActive(item.href), 'header__item--active-logo': index === 0 && isHomeActive}"
    )
      RouterLink(
        :to="item.href"
        class="header__link"
        :class="{'header__link--active': isActive(item.href), 'header__link--active-logo': index === 0 && isHomeActive}"
      ) {{ item.name }}
        Logo(v-if="index === 0 && item.href === '/'")
  .header__logo
    RouterLink(
      :to="menu[0].href"
      class="header__logo-link"
      :class="{'header__logo-link--active': isActive(menu[0].href)}"
    ) {{ menu[0].name }}
      Logo(v-if="menu[0].href === '/'")
  button.btn-reset.header__btn-burger(@click="handleToggleMenu")
    BurgerIcon(
      class="header__btn-burger-svg"
      :class="{'header__btn-burger-svg--active': isMenuOpen, 'header__btn-burger-svg--non-active': !isMenuOpen}"
    )
  .header__menu(
    v-show="isMenuOpen"
    :class="{'header__menu--open': isMenuOpen}"
    @wheel.prevent
    @touchmove.prevent
    @scroll.prevent
  )
    ul.list-reset.header__menu-list
      li(
          v-for="(item, index) in tabletMenu"
          :key="index"
          class="header__menu-item"
          :class="{'header__menu-item--active': isActive(item.href)}"
        )
          RouterLink(
            :to="item.href"
            class="header__menu-link"
            :class="{'header__menu-link--active': isActive(item.href)}"
          ) {{ item.name }}
</template>

<script lang="ts">
import Logo from "@/components/Common/Logo.vue";
import BurgerIcon from "@/components/Icon/BurgerIcon.vue";
import CrossIcon from "@/components/Icon/CrossIcon.vue";
import { defineComponent, PropType, computed, ref } from "vue";
import { RouterLink, useRoute } from "vue-router";
import { menuList } from "@/interfaces/IMenu";

export default defineComponent({
  name: "NavMenu",
  components: {
    Logo,
    RouterLink,
    BurgerIcon,
    CrossIcon,
  },
  props: {
    menu: {
      type: Array as PropType<menuList[]>,
      required: true,
    },
  },
  setup(props) {
    const route = useRoute();
    const isMenuOpen = ref<boolean>(false);
    const tabletMenu = props.menu.slice(1);

    const isHomeActive = computed(() => {
      return route.path === "/";
    });

    const isActive = (href: string) => {
      return route.path === href;
    };

    const handleToggleMenu = () => {
      isMenuOpen.value = !isMenuOpen.value;
    };

    return {
      isHomeActive,
      isActive,
      handleToggleMenu,
      isMenuOpen,
      tabletMenu,
    };
  },
});
</script>
<style lang="sass"></style>
