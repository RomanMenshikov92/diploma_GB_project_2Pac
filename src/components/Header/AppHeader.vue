<template lang="pug">
header.header
  .container.header__container
    NavMenu(:menu="menuList")
</template>

<script lang="ts">
import { defineComponent, computed } from "vue";
import NavMenu from "@/components/Header/NavMenu.vue";
import { useMenuStore } from "@/stores/Common/useMenuStore";

export default defineComponent({
  name: "AppHeader",
  components: {
    NavMenu,
  },
  setup() {
    const menuStore = useMenuStore();
    const menuList = computed(() => menuStore.getMenuItems);

    return {
      menuList,
    };
  },
});
</script>

<style lang="sass"></style>
