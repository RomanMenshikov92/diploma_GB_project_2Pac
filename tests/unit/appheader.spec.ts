import { shallowMount, mount } from "@vue/test-utils";
import AppHeader from "@/components/Header/AppHeader.vue";
import NavMenu from "@/components/Header/NavMenu.vue";
import { Pinia, createPinia } from "pinia";

jest.mock("@/components/Header/NavMenu.vue", () => ({
  name: "NavMenu",
  render: jest.fn(),
}));

describe("AppHeader.vue", () => {
  it("renders in AppHeader components: NavMenu", () => {
    const pinia: Pinia = createPinia();

    const wrapper = shallowMount(AppHeader, {
      global: {
        plugins: [pinia],
      },
    });
    expect(wrapper.findComponent(NavMenu).exists()).toBe(true);
  });

  it("renders correct Pug in header layout", () => {
    const pinia: Pinia = createPinia();

    const wrapper = mount(AppHeader, {
      global: {
        plugins: [pinia],
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });
});
