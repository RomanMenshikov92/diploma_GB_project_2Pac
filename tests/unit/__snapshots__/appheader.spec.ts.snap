// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppHeader.vue renders correct Pug in header layout 1`] = `
<header class="header">
  <div class="container header__container">
    <!---->
  </div>
</header>
`;
